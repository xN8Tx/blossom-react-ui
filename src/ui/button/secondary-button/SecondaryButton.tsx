import React from 'react';

import type { SecondaryButtonPropsType } from './SecondaryButton.types';

import style from './SecondaryButton.module.scss';

export function SecondaryButton({
  children,
  onClick,
  disabled = false,
  image,
  ...props
}: SecondaryButtonPropsType) {
  const isImage: boolean = image !== undefined;
  const isImageString = typeof image === 'string';

  return (
    <button
      className={style.Button}
      onClick={onClick}
      disabled={disabled}
      contains-image={isImage.toString()}
      {...props}
    >
      {isImage && (isImageString ? <img src={image} alt='Icon' /> : image!())}

      <span>{children}</span>
    </button>
  );
}
