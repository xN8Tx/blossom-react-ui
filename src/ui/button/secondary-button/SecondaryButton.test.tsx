import React from 'react';
import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import { SecondaryButton } from './SecondaryButton';

const mockOnClick = jest.fn();

describe('Secondary button test', () => {
  afterEach(() => cleanup());

  it('Should render', () => {
    render(
      <SecondaryButton onClick={mockOnClick}>Blossom Button</SecondaryButton>,
    );

    const button = screen.getByText('Blossom Button');

    expect(button).toBeInTheDocument();
  });
  it('Should click', () => {
    render(
      <SecondaryButton onClick={mockOnClick}>Blossom Button</SecondaryButton>,
    );

    const button = screen.getByText('Blossom Button');
    expect(button).toBeInTheDocument();

    fireEvent.click(button);

    expect(mockOnClick).toHaveBeenCalled();
  });
  it('Should not click', () => {
    render(
      <SecondaryButton onClick={mockOnClick} disabled={true}>
        Blossom Button
      </SecondaryButton>,
    );

    const button = screen.getByText('Blossom Button');
    expect(button).toBeInTheDocument();

    fireEvent.click(button);

    expect(mockOnClick).not.toHaveBeenCalled();
  });
  it('Snapshot test', () => {
    render(
      <SecondaryButton onClick={mockOnClick} disabled={true}>
        Blossom Button
      </SecondaryButton>,
    );

    const button = screen.getByText('Blossom Button');
    expect(button).toBeInTheDocument();

    expect(button).toMatchSnapshot();
  });
});
