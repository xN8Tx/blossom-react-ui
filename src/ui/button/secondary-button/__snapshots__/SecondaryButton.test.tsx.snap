// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secondary button test Snapshot test 1`] = `
<span>
  Blossom Button
</span>
`;
